media/docs/cpp/xe_slm_pipelined_gemm.md

```diff
@@ -0,0 +1,54 @@
+# Xe SLM Pipelined GEMM
+## Limitations of the L1 pipelined GEMM
+* Current GEMM implementation is not well-performed. SLM pipelined GEMM maybe can get better performance.
+
+## Goals
+
+The goal of introduing Shared Local Memory ([SLM](https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2025-2/shared-local-memory.html#id-d26320e84)) into pipelined GEMM, In more detail, we want to:
+
+* Fullly control of Shared local Memory
+  - Programmers take over and optimize shared memory path.
+  - Avoid competition between multiple work items.
+
+* Avoid bank conflict with cutlass component
+  - There is potential to use exsiting component (see `include/cute/swizzle.hpp`) in Cutlass.
+  - Using SLM as Cache, higher parallelism
+
+* Ease of gathering
+  - L1 cache is non-deterministic
+  - Gathering and reordering through controllable and high-bandwith memory
+
+## (Pros) Register->SLM->Register is more efficient than L1 path
+  - The key improvement is in the global -> register -> SLM part. The amount of data each thread needs to copy from global memory is smaller because threads collaborate to share the work of copying/reordering/dequantizing the data.
+  - Subgroup specification in each workgroup could avoid duplicated data tiles movemonet.
+  - In the current mainloop, each subgroup copies the same data from L1 -> register and does any necessary reorder/data conversion/dequantization on that data.
+  - To make an SLM flow efficient, we need heavy pipelining, typically triple or quadruple buffering in SLM.
+  - We can also combine the SLM flow with prefetch to L1, though prefetch is much less crucial in this case.
+  - Due to lack of asynchronous hardware features on Xe3p platform, programming model should be switched to Producer-Consumer pattern. Subgroups producers only copy from global memory to SLM.
```

Comment:
It isn't at all clear that producer-consumer would be beneficial. Probably needs experiments or a good analysis based on prior data.